src/main/java/appeng/tile/crafting/TileCraftingStorageTile.java

```diff
@@ -36,24 +38,29 @@ protected ItemStack getItemFromTile( final Object obj )
 		final IBlocks blocks = AEApi.instance().definitions().blocks();
 		final int storage = ( (TileCraftingTile) obj ).getStorageBytes() / KILO_SCALAR;
 
+		Optional<ItemStack> is;
+
 		switch( storage )
 		{
 			case 4:
-				for( final ItemStack stack : blocks.craftingStorage4k().maybeStack( 1 ).asSet() )
+				is = blocks.craftingStorage4k().maybeStack( 1 );
+				if( is.isPresent() )
 				{
-					return stack;
+					return is.get();
```

Comment:
These could be replaced with `return maybeStack( 1 ).orElse( null )`. The `break`s will prevent any further case and finally it always returns null